curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/646d404259f512c1a9e86ac1.md

```diff
@@ -7,11 +7,11 @@ dashedName: step-74
 
 # --description--
 
-In your `highPrecedence` function, declare a `regex` variable. Assign it a regular expression that matches a number (including decimal numbers) followed by a `*` or `/` operator followed by another number.
+In your `highPrecedence` function, you should use a regex to check if the provided string matches the pattern of a number followed by a `*` or `/` operator followed by another number. 
```

Comment:
The requirements are missing the `regex` variable and that it has to be a `const`. If it is required, it should be part of the requirements.

Possible suggestion:

```suggestion
In your `highPrecedence` function, declare a `regex` variable using `const` and assign it a regex that checks if the string passed to the `str` parameter matches the pattern of a number followed by a `*` or `/` operator followed by another number. 
```

Reply:
Is it possible to simplify it in such a way I don't need to check if the camper is using the exact same name of the regex without sacrificing too much?

Reply:
You can replace `regex` with `\w+` in the tests, and then remove the reference to `regex` as the variable name. We should keep the `const` requirement.

````md
In your `highPrecedence` function, declare a variable using `const` and assign it a regex that checks if the string passed to the `str` parameter matches the pattern of a number followed by a `*` or `/` operator followed by another number.

Your function should return a boolean value. Remember that you can use the `test()` method for this.

Each number, and the operator, should be in separate capture groups.

# --hints--

You should declare a variable in your `highPrecedence` function for your regex.

```js
assert.match(code, /const\s+highPrecedence\s*=\s*(\(\s*str\s*\)|str)\s*=>\s*{\s*(?:const|let|var)\s+\w+/);
```

You should use `const` to declare your regex variable.

```js
assert.match(code, /const\s+highPrecedence\s*=\s*(\(\s*str\s*\)|str)\s*=>\s*{\s*const\s+\w+/);
```

Your regex variable should contain a regular expression.

```js
assert.match(code, /const\s+highPrecedence\s*=\s*(\(\s*str\s*\)|str)\s*=>\s*{\s*const\s+\w+\s*=\s*\//);
```

Your `highPrecedence` function should return a boolean value.

```js
assert.isBoolean(highPrecedence("12*2"));
```

Your `highPrecedence` function should correctly check if the string matches the pattern of a number followed by a `*` or `/` operator followed by another number.

```js
assert.isTrue(highPrecedence("5*3"));
assert.isFalse(highPrecedence("5"));
assert.isTrue(highPrecedence("10/2"));
assert.isFalse(highPrecedence("*"));
```
````

Reply:
Thank you for the additional input.